### ISSUE TITLE
SdesTransport doesn't work and should be removed

### ISSUE THREAD

**Issue by jnoring** [lynckia/licode#183]
Troubleshooting a deadlock in erizo, I noticed:

Thread 27 (Thread 0x7f67b2ffd700 (LWP 20675)):
#0  0x00007f67e474b89c in __lll_lock_wait () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x00007f67e4747065 in _L_lock_858 () from /lib/x86_64-linux-gnu/libpthread.so.0
#2  0x00007f67e4746eba in pthread_mutex_lock () from /lib/x86_64-linux-gnu/libpthread.so.0
#3  0x00007f67dfbdfa59 in boost::mutex::lock (this=0x21baa88) at /usr/include/boost/thread/pthread/mutex.hpp:52
#4  0x00007f67dfbe07a4 in boost::unique_lockboost::mutex::lock (this=0x7f67b2ff5160) at /usr/include/boost/thread/locks.hpp:412
#5  0x00007f67dfbe0309 in boost::unique_lockboost::mutex::unique_lock (this=0x7f67b2ff5160, m_=...) at /usr/include/boost/thread/locks.hpp:290
#6  0x00007f67dfc33b0e in erizo::WebRtcConnection::queueData (this=0x21ba800, comp=1, buf=0x6cc96f0 "\200", <incomplete sequence \311>, length=22, transport=0x21be200)

```
at /mnt/opt/hirevue/build/workspace/licode/erizo/src/erizo/WebRtcConnection.cpp:418
```
#7  0x00007f67dfc20ea1 in erizo::SdesTransport::write (this=0x21be200, data=0x53cc554 "\200", <incomplete sequence \311>, len=8)

```
at /mnt/opt/hirevue/build/workspace/licode/erizo/src/erizo/SdesTransport.cpp:118 
```
#8  0x00007f67dfc340a3 in erizo::WebRtcConnection::sendLoop (this=0x21ba800) at /mnt/opt/hirevue/build/workspace/licode/erizo/src/erizo/WebRtcConnection.cpp:482
#9  0x00007f67dfc3724e in boost::_mfi::mf0<void, erizo::WebRtcConnection>::operator() (this=0x53cccc8, p=0x21ba800) at /usr/include/boost/bind/mem_fn_template.hpp:49
#10 0x00007f67dfc371be in boost::_bi::list1boost::_bi::value<erizo::WebRtcConnection* >::operator()<boost::_mfi::mf0<void, erizo::WebRtcConnection>, boost::_bi::list0> (this=0x53cccd8, f=..., a=...)

```
at /usr/include/boost/bind/bind.hpp:253
```
#11 0x00007f67dfc3716d in boost::_bi::bind_t<void, boost::_mfi::mf0<void, erizo::WebRtcConnection>, boost::_bi::list1boost::_bi::value<erizo::WebRtcConnection* > >::operator() (this=0x53cccc8)

```
at /usr/include/boost/bind/bind_template.hpp:20
```

---Type <return> to continue, or q <return> to quit---
#12 0x00007f67dfc37132 in boost::detail::thread_data<boost::_bi::bind_t<void, boost::_mfi::mf0<void, erizo::WebRtcConnection>, boost::_bi::list1boost::_bi::value<erizo::WebRtcConnection* > > >::run

```
(this=0x53ccb40) at /usr/include/boost/thread/detail/thread.hpp:61
```
#13 0x00007f67df36cce9 in thread_proxy () from /usr/lib/libboost_thread.so.1.46.1
#14 0x00007f67e4744e9a in start_thread () from /lib/x86_64-linux-gnu/libpthread.so.0
#15 0x00007f67e447231d in clone () from /lib/x86_64-linux-gnu/libc.so.6
#16 0x0000000000000000 in ?? ()

Basically, the thread is deadhung on itself trying to lock WebRtcConnection::receiveVideoMutex_, which is not a recursive mutex.

Do we even still need this file given that both FF and Chrome are now dtls?  If no, I'm happy to remove this.  Otherwise it should probably be fixed.
